PWGJE/TableProducer/jettaggerhfExtension.cxx

```diff
@@ -0,0 +1,197 @@
+
+  // Operation and minimisation criteria
+  Configurable<bool> doFillHistogram{"doFillHistogram", false, "fill histogram"};
+  ConfigurableAxis binDcaX{"binDcaX", {1001, -1.5f, 1.5f}, ""};
+  ConfigurableAxis binDcaY{"binDcaY", {1001, -1.5f, 1.5f}, ""};
+  ConfigurableAxis binDcaXY{"binDcaXY", {1001, -1.5f, 1.5f}, ""};
+  ConfigurableAxis binDcaZ{"binDcaZ", {1001, -1.5f, 1.5f}, ""};
+  ConfigurableAxis binDcaXYZ{"binDcaXYZ", {1001, -1.5f, 1.5f}, ""};
+  ConfigurableAxis binDcaXYSigma{"binDcaXYSigma", {1001, -0.5f, 1.5f}, ""};
+  ConfigurableAxis binDcaXYZSigma{"binDcaXYZSigma", {1001, -0.5f, 1.5f}, ""};
+
+  // Axis
+  AxisSpec DcaXAxis = {binDcaX, "DCA_{X} [#mum]"};
+  AxisSpec DcaYAxis = {binDcaY, "DCA_{Y} [#mum]"};
+  AxisSpec DcaXYAxis = {binDcaXY, "DCA_{XY} [#mum]"};
+  AxisSpec DcaZAxis = {binDcaZ, "DCA_{Z} [#mum]"};
+  AxisSpec DcaXYZAxis = {binDcaXYZ, "DCA_{XYZ} [#mum]"};
+  AxisSpec DcaXYSigmaAxis = {binDcaXY, "#sigma_{DCA_{XY}} [#mum]"};
+  AxisSpec DcaXYZSigmaAxis = {binDcaXY, "#sigma_{DCA_{XYZ}} [#mum]"};
+
+  int runNumber = -1;
+
+  HistogramRegistry registry{"registry", {}, OutputObjHandlingPolicy::AnalysisObject};
+
+  void init(o2::framework::InitContext& initContext)
+  {
+    ccdb->setURL(ccdburl);
+    ccdb->setCaching(true);
+    ccdb->setLocalObjectValidityChecking();
+    lut = o2::base::MatLayerCylSet::rectifyPtrFromFile(ccdb->get<o2::base::MatLayerCylSet>(lutPath));
+
+    if (doFillHistogram) {
+      registry.add("hDCAx", "DCA to PV; DCA_{x} (#mum)", {HistType::kTH1F, {DcaXAxis}});
+      registry.add("hDCAy", "DCA to PV; DCA_{y} (#mum)", {HistType::kTH1F, {DcaYAxis}});
+      registry.add("hDCAz", "DCA to PV; DCA_{z} (#mum)", {HistType::kTH1F, {DcaZAxis}});
+      registry.add("hDCAxy", "DCA to PV; DCA_{xy} (#mum)", {HistType::kTH1F, {DcaXYAxis}});
+      registry.add("hDCAxyz", "DCA to PV; DCA_{xyz} (#mum)", {HistType::kTH1F, {DcaXYZAxis}});
+      registry.add("hDCAxyVsDCAxyFromExt", "DCA to PV; DCA_{xy} (#mum); DCA_{xy}^{ext} (#mum)", {HistType::kTH2F, {{DcaXYAxis}, {DcaXYAxis}}});
+      registry.add("hDCAxy_uncertainty", "uncertainty of DCA to PV; #sigma_{DCA_{xy}} (#mum)", {HistType::kTH1F, {DcaXYSigmaAxis}});
+      registry.add("hDCAxyz_uncertainty", "uncertainty of DCA to PV; #sigma_{DCA_{xyz}} (#mum)", {HistType::kTH1F, {DcaXYZSigmaAxis}});
+    }
+  }
+
+  template <typename T>
+  void initCCDB(T const& bc)
+  {
+    if (runNumber == bc.runNumber()) {
+      return;
+    }
+    grpmag = ccdb->getForTimeStamp<o2::parameters::GRPMagField>(grpmagPath, bc.timestamp());
+    LOG(info) << "Setting magnetic field to current " << grpmag->getL3Current() << " A for run " << bc.runNumber() << " from its GRPMagField CCDB object";
+    o2::base::Propagator::initFieldFromGRP(grpmag);
+    o2::base::Propagator::Instance()->setMatLUT(lut);
+    mVtx = ccdb->getForTimeStamp<o2::dataformats::MeanVertexObject>(mVtxPath, bc.timestamp());
```

Comment:
it seems in this function you only need this line?

Reply:
I just copy from Common/TableProducer/trackPropagation.cxx
it doesn't need to set magnetic field from ccdb?